packages/lwc/types.d.ts

```diff
@@ -220,12 +220,28 @@ declare module 'lwc' {
         consumerDisconnectedCallback?: (consumer: ContextConsumer) => void;
     }
 
-    interface WireAdapterConstructor {
-        new (callback: WireDataCallback): WireAdapter;
+    interface WireAdapterConstructor<T> {
+        new (callback: WireDataCallback): WireAdapter<T>;
         configSchema?: Record<string, WireAdapterSchemaValue>;
         contextSchema?: Record<string, WireAdapterSchemaValue>;
     }
 
     type Contextualizer = (elm: EventTarget, options: ContextProviderOptions) => void;
-    export function createContextProvider(config: WireAdapterConstructor): Contextualizer;
+    export function createContextProvider<T>(config: WireAdapterConstructor<T>): Contextualizer;
+}
+
+declare interface WiredValue<T extends (...args: any) => any> {
+    error?: FetchResponse;
+    data?: Awaited<ReturnType<T>>;
+}
+
+declare interface FetchResponse<T = any> {
+    // The body of the response, which is defined by the underlying API.
+    body: T | Array<T>;
+    // Specifies whether the response was successful or not. For an error, ok is always false and contains a status in the range 400–599.
+    ok: boolean;
+    // Contains the status code of the response, for example, 404 if a resource is not found or 500 for an internal server error.
+    status: number;
+    // Contains the status message corresponding to the status code, for example, NOT_FOUND for a status code of 404.
+    statusText: string;
 }
```

Comment:
If I'm not mistaken, this looks like a DOM [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object? Can it be imported from TypeScript's `dom.d.ts` instead?

Reply:
I think it is really similar, and I also don't think I have all the details. If I look at the Lightning Data Service docs I can get some idea of what the response will be, that is where I'm pulling the shape from. It shares a similar definition but I don't think they are specifically the same.
https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.data_error

I wouldn't be surprised to find out this isn't a generic enough shape for this level and we need to support different definitions at a per-adapter level, where the adapter would somehow provide the type shape of it's possible error response. Would appreciate additional thoughts or a recommendation to pull it for the time being until we can gather clearer guidance.

Reply:
Removed, per @pmdartus